scripts/graphql-codegen/generator.ts

```diff
@@ -0,0 +1,141 @@
+  {
+    name: "PushMessages",
+    searchKey: "PushMessages",
+    apiPath: "client-app/core/api/graphql/push-messages",
+  },
+];
+
+const GENERAL_CONFIG = {
+  dedupeFragments: true,
+  identifierName: "OperationNames",
+  maybeValue: "T",
+  scalars: {
+    BigInt: "number",
+    Byte: "number",
+    Date: "string",
+    DateOnly: "string",
+    Decimal: "number",
+    DynamicPropertyValue: "string | number | boolean | null",
+    Guid: "string",
+    Half: "number",
+    Long: "number",
+    Milliseconds: "number",
+    ModuleSettingValue: "string | number | boolean | null",
+    OptionalDecimal: "number | undefined",
+    OptionalNullableDecimal: "number | null | undefined",
+    OptionalString: "string | undefined",
+    PropertyValue: "string | number | boolean | null",
+    SByte: "number",
+    Seconds: "number",
+    Short: "number",
+    TimeOnly: "string",
+    UInt: "number",
+    ULong: "number",
+    Uri: "string",
+    UShort: "number",
+  },
+  skipTypename: true,
+  useTypeImports: true,
+  skipGraphQLImport: true,
+};
+
+const GENERAL_PLUGINS = [
+  {
+    add: {
+      content: "// This file is auto-generated. Do not edit manually.\n",
+    },
+  },
+  "typescript",
+  "typescript-operations",
+  "typed-document-node",
+  "named-operations-object",
+];
+
+async function runCodegen() {
+  // eslint-disable-next-line no-console
+  console.log("\nGenerate types for general modules:");
+  const typesPath = `${allModulesPath}/types.ts`;
+  await generate(
+    {
+      schema: SCHEMA_PATH,
+      documents: [
+        addExtension(allModulesPath),
+        // exclude independent modules from general modules
+        ...independentModules.map((module) => `!${addExtension(module.apiPath)}`),
+      ],
+      generates: {
+        [typesPath]: {
+          plugins: GENERAL_PLUGINS,
+          config: GENERAL_CONFIG,
+        },
+      },
+    },
+    true,
+  );
+  // eslint-disable-next-line no-console
+  console.log(`Types for general modules have been generated in "${typesPath}"`);
+
+  const JSONString = await readJsonAndReturnString(SCHEMA_PATH);
+
+  const installedIndependentModules = independentModules.filter((el) => JSONString?.includes(el.searchKey));
+
+  if (installedIndependentModules.length) {
+    // eslint-disable-next-line no-console
+    console.log("\nGenerate types for independent modules:");
+  }
+
+  for (let i = 0; i < installedIndependentModules.length; i++) {
+    const module = independentModules[i];
+    const moduleTypesPath = `${module.apiPath}/types.ts`;
+    await generate(
+      {
+        schema: SCHEMA_PATH,
+        documents: addExtension(module.apiPath),
+        generates: {
+          [moduleTypesPath]: {
+            plugins: GENERAL_PLUGINS,
+            config: GENERAL_CONFIG,
+          },
+        },
+      },
+      true,
+    );
+    // eslint-disable-next-line no-console
+    console.log(`Types for "${module.name}" module have been generated in "${moduleTypesPath}"`);
+  }
```

Comment:
This will cause duplicate types. See https://github.com/VirtoCommerce/vc-theme-b2b-vue/blob/feat/VCST-626-graphql-multiple-files/scripts/generate-graphql-bundle.ts approach:
1. Generate separate files for all
2. Create files which imports them for backward-compatibility or to make separate module bundle

Reply:
duplicates if ok

Reply:
It will not require you to cast all types which are duplicated, but also increase bundle size